### ISSUE TITLE
Streamline section changes (questionnaire navigation) and how they are logged

### ISSUE THREAD

**Issue by tahini** [chairemobilite/evolution#742]
Currently, when clicking on a section's `submit` button, the button widget needs to call an action that will automatically validate the current section and navigate to the next section as per the section's configuration. That next section will thus be considered as started.

But many surveys have the kind of workflow where we loop through a few sections for each household member, before actually reaching the last sections. In this case, in this the last section's preload function that needs to tell the survey to change the active section, by setting the `_activeSection` value to where it needs to go.

This puts the responsibilty of the section navigation to individual survey's section preload, and it also has the consequence of setting the last section as started, even if the participant was never in this section.

The section navigation in the survey should be handled by some API who should figure out what the next section is and the logs should only consider a section started if the participant actually saw the page.

**Comment by tahini** [comment]
This will be a major breaking change for current surveys, but it needs to be done. Should we tag a branch `legacy` (or `v.0.2.2` as we currently have) for our current surveys where we can backport compatible changes and make a totally divergent `main` branch? And maybe change that famous 0.2.2 version that we have in all our package.json files?

**Comment by greenscientist** [comment]
Yeah, we probably abuse a bit too much the term "legacy" in our projects. But tagging (in branching if we need it) is probably a good option. 

Not fully sure what's the best version scheme.   Maybe we can use the opportunity to bump the current version to 0.3.  When you do your breaking change, we can bump it to 0.5 (Yeah, I ski the .4, I want to mark the break!).   After that, we should take the habit of increasing the number (0.6, 0.7, 0.8) for each breaking change like this one. No need to version everything thing, but that will give a quick idea that a specific version of the survey will need major rework based on it's version number. This would me mostly informative, as the actual version tracking is done with the submodule hash version.

**Comment by tahini** [comment]
I foresee there will be a truck load of breaking changes in the upcoming weeks/months, so we might as well branch a `v0.3` version (for the sake of the currently running surveys), then, when we do the first major breaking change (so badly breaking that it would be a mess to upgrade current surveys), we'll upgrade `main` to 0.5 and we say we don't use it in production until it's "ready".

Does that sound like a plan?

Other option is to keep `main` production ready, and I do the proposed changes in a `v0.5` branch. Ultimately, the result is the same, it's just the status of the `main` branch that changes.

**Comment by greenscientist** [comment]
We don't have new survey foreseen (I know we can get some at any moment), so I Think going with the first plan you described sounds good. If we have a new survey and we want to use the new stuff, we could branch out a 0.5 at that point and move main to "0.6"
That probably gives us a few months of "free" developpment.
